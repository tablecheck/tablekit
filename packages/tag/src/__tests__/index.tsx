import { mount } from '@tablecheck/tablekit-enzyme';
import '@tablecheck/tablekit-free-icon-config';

import { Tag } from '../index';

beforeEach(() => {
  global.console.error = jest.fn();
});

describe('Tag', () => {
  test('should be possible to create a component', () => {
    const wrapper = mount(<Tag />);
    expect(wrapper).not.toBeUndefined();
  });

  test('should correctly apply styling', () => {
    expect(mount(<Tag />)).toMatchSnapshot();
  });

  test('should allow passing of custom className', () => {
    expect(mount(<Tag className="test-class" />)).toMatchSnapshot();
  });
});
