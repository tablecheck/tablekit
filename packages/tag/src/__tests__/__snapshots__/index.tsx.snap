// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should allow passing of custom className 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: rgba(0, 0, 0, 0.11);
  color: #323232;
  background: transparent;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  border-width: 1px;
  border-style: solid;
  box-sizing: border-box;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: 0;
  max-width: 100%;
  outline: none;
  pointer-events: auto;
  text-align: center;
  white-space: nowrap;
  width: auto;
  vertical-align: middle;
  border-radius: 8px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  padding: 0 4px;
  height: 32px;
  font-weight: bold;
  -webkit-transition: 120ms ease-in-out;
  transition: 120ms ease-in-out;
}

.emotion-0:hover {
  background: rgba(0, 0, 0, 0.05);
}

.emotion-0 .svg-inline--fa {
  width: 16px;
  height: 16px;
}

.emotion-0::-moz-focus-inner {
  border: 0;
  margin: 0;
  padding: 0;
}

.emotion-0::after {
  border-radius: 8px;
  -webkit-transition: 120ms ease-in-out;
  transition: 120ms ease-in-out;
}

.emotion-2 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  position: relative;
}

<Tag
  className="test-class"
>
  <StyledTag
    appearance="outline"
    className="test-class"
    isSelected={false}
    size="regular"
  >
    <Noop />
    <button
      className="test-class emotion-0 emotion-1"
      size="regular"
    >
      <TagWrapper>
        <Noop />
        <span
          className="emotion-2 emotion-3"
        />
      </TagWrapper>
    </button>
  </StyledTag>
</Tag>
`;

exports[`Tag should correctly apply styling 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  border-color: rgba(0, 0, 0, 0.11);
  color: #323232;
  background: transparent;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  border-width: 1px;
  border-style: solid;
  box-sizing: border-box;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: 0;
  max-width: 100%;
  outline: none;
  pointer-events: auto;
  text-align: center;
  white-space: nowrap;
  width: auto;
  vertical-align: middle;
  border-radius: 8px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  padding: 0 4px;
  height: 32px;
  font-weight: bold;
  -webkit-transition: 120ms ease-in-out;
  transition: 120ms ease-in-out;
}

.emotion-0:hover {
  background: rgba(0, 0, 0, 0.05);
}

.emotion-0 .svg-inline--fa {
  width: 16px;
  height: 16px;
}

.emotion-0::-moz-focus-inner {
  border: 0;
  margin: 0;
  padding: 0;
}

.emotion-0::after {
  border-radius: 8px;
  -webkit-transition: 120ms ease-in-out;
  transition: 120ms ease-in-out;
}

.emotion-2 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  max-width: 100%;
  position: relative;
}

<Tag>
  <StyledTag
    appearance="outline"
    isSelected={false}
    size="regular"
  >
    <Noop />
    <button
      className="emotion-0 emotion-1"
      size="regular"
    >
      <TagWrapper>
        <Noop />
        <span
          className="emotion-2 emotion-3"
        />
      </TagWrapper>
    </button>
  </StyledTag>
</Tag>
`;
